pkg/capability/capability.go

```diff
@@ -0,0 +1,135 @@
+package capability
+
+import (
+	"sort"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+type ClusterCapabilities struct {
+	KnownCapabilities             map[configv1.ClusterVersionCapability]struct{}
+	EnabledCapabilities           map[configv1.ClusterVersionCapability]struct{}
+	ImplicitlyEnabledCapabilities []configv1.ClusterVersionCapability
+}
+
+// GetImplicitlyEnabledCapabilities filters out from a set of capabilities
+// the ones that
+// - are not in the list enabledCapabilities of clusterCapabilities
+// - are not the keys of clusterCapabilities.EnabledCapabilities
+// - are not in the list clusterCapabilities.ImplicitlyEnabledCapabilities
+// The returned list is the implicitly enabled capabilities.
+func GetImplicitlyEnabledCapabilities(enabledCapabilities []configv1.ClusterVersionCapability,
+	capabilities []configv1.ClusterVersionCapability,
+	clusterCapabilities ClusterCapabilities) []configv1.ClusterVersionCapability {
```

Comment:
TBH the signature and the godoc does not help too much in understanding what is the purpose of the method and what (high-level, semantic) operation ti performs. You give it some capabilities, and it filters some out. It is supposed to return implicitly enabled capabilities, but at the same time, it filters the content of `clusterCapabilities.ImplicitlyEnabledCapabilities` **out**? That's confusing. Also, if I am a caller, how is `enabledCapabilities` different from `clusterCapabilities.EnabledCapabilities`?

Also, they are all capabilities, so calling the variables as such does not help much. Could we call them `enabled` and `input` or something that carries more semantics?